ngraph/frontend/onnx/onnx_common/src/onnx_model_validator.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "onnx_common/onnx_model_validator.hpp"
+
+#include <algorithm>
+#include <array>
+#include <exception>
+#include <map>
+#include <vector>
+
+namespace detail {
+namespace onnx {
+enum Field {
```

Comment:
enum class?

Reply:
In this PR I've just moved the validator to other place, but as I remember enum classes were not used here to avoid many static_casts (especially in `PbKey` processing) which makes the code harder to read.